New evict and recover techniques

ryantxu1 opened the pull request:
New Evict Techniques:

- Evict Access
    - Unenroll Suspicious MFA Tokens
    - Remove User Account
- Evict From Network
    - Black Hole
    - Flush DNS Cache
- Evict Object
    - Remove Registry Key

New Restore Techniques

ryantxu1 commented:
@netfl0 STILL WIP! But if time allows, please review to see if I'm on the right track

netfl0 commented on src/ontology/d3fend-protege.ttl:
```diff
@@ -4016,18 +4016,6 @@ Effective implementation requires identifying any location that could end up con
     :kb-reference :Reference-AllLoginsSinceLastBoot_MITRE,
         :Reference-SystemsAndMethodsForDetectingCredentialTheft_SymantecCorp .
 
-:CredentialEviction a :CredentialEviction,
```
This should definitely remain. Think in terms of a proper taxonomy versus what the matrix looks like.

netfl0 commented on src/ontology/d3fend-protege.ttl:
```diff
@@ -2701,6 +2701,19 @@ BERT (language model). (n.d.). In Wikipedia. [Link](https://en.wikipedia.org/wik
     :kb-reference :Reference-TokenlessBiometricTransactionAuthorizationMethodAndSystem,
         :Reference-www.biometric-solutions.com_keystroke-dynamics .
 
+:BlackHole a :BlackHole,
```
Might want to use BlackHole**Route**

netfl0 commented on src/ontology/d3fend-protege.ttl:
```diff
@@ -16713,6 +16778,30 @@ Analysis algorithms look for patterns in the network traffic captured from the s
     :definition "A removable media device is a hardware device used for computer storage and that is designed to be inserted and removed from the system.  It is distinct from other removable media in that all the hardware required to read the data are built into the device.  So USB flash drives and external hard drives are removable media devices, whereas tapes and disks are not, as they require  ...
     rdfs:seeAlso <http://dbpedia.org/resource/Removable_media> .
 
+:RemoveRegistryKey a owl:Class,
```
should be DeleteRegistryKey

netfl0 commented on src/ontology/d3fend-protege.ttl:
```diff
@@ -16713,6 +16778,30 @@ Analysis algorithms look for patterns in the network traffic captured from the s
+    :definition "Delete a registry key." ;
+    :kb-reference :Reference-CybersecurityIncidentandVulnerabilityResponsePlaybooks .
+
+:RemoveUserAccount a owl:Class,
```
Should be delete

netfl0 commented on src/ontology/d3fend-protege.ttl:
```diff
@@ -17441,6 +17544,12 @@ Yu, L., Zhang, W., Wang, J., & Yu, Y. (2017). SeqGAN: Sequence Generative Advers
 
+:ServiceAccount a owl:Class ;
+    rdfs:label "Service Account" ;
+    rdfs:subClassOf :DigitalInformationBearer ;
```
should rather be a sublcass of d3f:UserAccount

netfl0 commented on src/ontology/d3fend-protege.ttl:
```diff
@@ -2701,6 +2713,19 @@ BERT (language model). (n.d.). In Wikipedia. [Link](https://en.wikipedia.org/wik
     :kb-reference :Reference-TokenlessBiometricTransactionAuthorizationMethodAndSystem,
         :Reference-www.biometric-solutions.com_keystroke-dynamics .
 
+:BlackHoleRoute a :BlackHoleRoute,
```
This might be its own taxonomy. Sometimes you might route for more monitoring/inspection.

ryantxu1 replied:
yeah I agree that "BlackHoleRoute" might be too specific for a technique, will broaden the scope...

netfl0 commented on src/ontology/d3fend-protege.ttl:
```diff
@@ -14481,6 +14551,18 @@ Wikipedia. (n.d.). Nonlinear regression. [Link](https://en.wikipedia.org/wiki/No
     :d3fend-id "D3A-NPM" ;
     :definition "Numeric pattern matching uses a pattern specification and sees if the numeric value matches that pattern--simple forms include exact matching and range matching." .
 
+:ObjectEviction a :ObjectEviction,
```
We DO need a broader category like this, Object is something specific in CCO however. Would like to get thoughts from @hack-sentinel @giacomodecolle  and @johnbeve

Object will resonate with practitioners but has ontological issues.

We already use "Decoy Object", but I am deciding if we want to exacerbate this or move in another direction.

giacomodecolle replied:
Yes, "object" is a material entity in BFO/CCO. Have you considered using two labels, one for practitioners and one for ontologists? I think @johnbeve usually recommends prefLabel or altLabel. 

Then we can also start thinking about a corresponding CCO mapping and label (e.g. "malicious information entity eviction").

ryantxu1 replied:
@hack-sentinel @netfl0 I guess this is also related to how the current "Restore" tactic is set up since it has a "Restore Object" bucket. Personally, I'm in favor of this because otherwise we get into a situation where there are potentially too many buckets (Evict Hardware Object, Evict Software, etc)

What do yall think? Maybe we can use "Evict ARTIFACT" instead?

netfl0 reviewed (changes requested):
Need feedback from others listed.

ryantxu1 commented:
Before 8/16 commit:

![image](https://github.com/user-attachments/assets/81a378a0-73a8-4e9c-91c2-8f26f4374f67)

After:

![image](https://github.com/user-attachments/assets/299f3374-2f8e-482f-8680-fda4bd2475d5)


**Notes**
- Added _Session Termination_ and _Takedown Domain Registration_ techniques latter courtesy of #199 
- Deleted Black Hole Route
  - Blackholing and its cousin sinkholing are captured by DNS Denylisting
 - Deleted Network Eviction
   - Given the matrix heuristics, there's not enough techniques that fall under that currently, so Flush DNS Cache and Takedown Domain Registration are moved to object eviction. (The definition of Evict still covers 'network')
   
 @netfl0 Please review

ryantxu1 commented:
I could see Rogue Device Removal as a possible addition... @apapaa do you have any further details or references on this?

netfl0 commented on src/ontology/d3fend-protege.ttl:
```diff
@@ -4020,13 +4032,10 @@ Effective implementation requires identifying any location that could end up con
-        [ a owl:Restriction ;
-            owl:onProperty :enables ;
-            owl:someValuesFrom :Evict ] ;
+    rdfs:subClassOf :AccessEviction ;
```
this doesn't seem right, AccessMediator is a type of DA in the current model. Not a technique.

ryantxu1 replied:
Oh, thats strange, I didn't make that change manually - I agree though, will revert

ryantxu1 replied:
@netfl0 Upon further review, I don't think Access Mediator was actually changed. The line refers to Credential Eviction becoming a subclass of Access Eviction lol

netfl0 commented:
Edits:

- [ ] Drop "suspicious" from unenroll; more importantly MFA Token Disenrollment 
- [ ] Move process Eviction & Credential back up, and reassociate with tactic
- [ ] Move Delete User account to Object Eviction
   - [ ]  User Account Deletion versus Delete User
- [ ] Move Session Termination to Process Eviction
- [ ] Delete Access Eviction; bring back it later as we expand IAM modeling
- [ ] instead, Domain Registration Takedown
- [ ] DNS Cache Eviction
- [ ] Registry Key Deletion

apapaa commented:
> Edits:
> 
> * [x]  Drop "suspicious" from unenroll; more importantly MFA Token Disenrollment
> * [x]  Move process Eviction & Credential back up, and reassociate with tactic
> * [x]  Move Delete User account to Object Eviction
>   
>   * [x]   User Account Deletion versus Delete User
> * [x]  Move Session Termination to Process Eviction
> * [x]  Delete Access Eviction; bring back it later as we expand IAM modeling
> * [x]  instead, Domain Registration Takedown
> * [ ]  DNS Cache Eviction
> * [ ]  Registry Key Deletion

@ryantxu1  and @netfl0 
Agree with "Domain Registration Takedown" wording. Could also add the following:
- "Service Takedown" - related to takedown requests that could be sent to abuse notification emails for hosting providers, DNS providers and SaaS providers

ryantxu1 commented:
Post restructure:

![image](https://github.com/user-attachments/assets/0d9b3392-4e25-4266-855e-0316107534de)

apapaa commented:
> Post restructure:
> 
> ![image](https://private-user-images.githubusercontent.com/33180883/359221593-0d9b3392-4e25-4266-855e-0316107534de.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjQxMDA4MDQsIm5iZiI6MTcyNDEwMDUwNCwicGF0aCI6Ii8zMzE4MDg4My8zNTkyMjE1OTMtMGQ5YjMzOTItNGUyNS00MjY2LTg1NWUtMDMxNjEwNzUzNGRlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MTklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODE5VDIwNDgyNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTAyZDkzNjcwNzZhN2Y5YzkxMzU4NGM4NmZkNWFhNjNkMWE3YzkyZDgxNmU1OGQ4YjNmMzliNjdjZmNhN2NjMGYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.OsSF8HWRTp0BgysupZJr_haERyzhWjMK8L-lr57SOFE)

Is Revocation a better word than Disenrollment under "MFA Token Disenrollment"

netfl0 commented:
> > Post restructure:
> > ![image](https://private-user-images.githubusercontent.com/33180883/359221593-0d9b3392-4e25-4266-855e-0316107534de.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjQxMDA4MDQsIm5iZiI6MTcyNDEwMDUwNCwicGF0aCI6Ii8zMzE4MDg4My8zNTkyMjE1OTMtMGQ5YjMzOTItNGUyNS00MjY2LTg1NWUtMDMxNjEwNzUzNGRlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MTklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODE5VDIwNDgyNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTAyZDkzNjcwNzZhN2Y5YzkxMzU4NGM4NmZkNWFhNjNkMWE3YzkyZDgxNmU1OGQ4YjNmMzliNjdjZmNhN2NjMGYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.OsSF8HWRTp0BgysupZJr_haERyzhWjMK8L-lr57SOFE)
> 
> Is Revocation a better word than Disenrollment under "MFA Token Disenrollment"

I think so. I think we're trying to avoid the notion of getting it physically returned. However, CRLs use the same terminology. In D3FEND 1.0 we should also change CredentialRevoking to CredentialRevocation . cc @hack-sentinel

ryantxu1 commented:
Renaming complete

netfl0 commented on src/ontology/d3fend-protege.ttl:
```diff
@@ -4046,10 +4046,10 @@ Effective implementation requires identifying any location that could end up con
     rdfs:isDefinedBy <http://dbpedia.org/resource/Credential_Management> ;
     :definition "Credential Management, also referred to as a Credential Management System (CMS), is an established form of software that is used for issuing and managing credentials as part of public key infrastructure (PKI)." .
 
-:CredentialRevoking a :CredentialRevoking,
```
I meant to do this in the future, but we can use this as a case study to test out our deprecation strategy. cc @hack-sentinel

ryantxu1 commented:
Are there any other changes needed before this is pushed? @netfl0

ryantxu1 commented:
@netfl0  is there anything else to be done before merging?

ryantxu1 commented:
![image](https://github.com/user-attachments/assets/758f5420-0017-44a3-93a8-ea7f927f8942)

apapaa commented:
I think it looks good @ryantxu1
@netfl0 could you please consider adding me as a Contributor when this goes live?

netfl0 commented:
it appears there are merge conflicts, can you resolve? @ryantxu1
